dissect/target/tools/reg.py

```diff
@@ -38,20 +44,31 @@ def main():
 
     try:
         for target in Target.open_all(args.targets):
+            if not target.has_function("registry"):
+                log.error("Target %s has no Windows Registry", target)
+                continue
+
             try:
-                if args.value:
-                    for key in target.registry.keys(args.key):
-                        try:
-                            print(key.value(args.value))
-                        except RegistryError:
-                            continue
-                else:
+                keys = list(target.registry.keys(args.key))
+
+                if not keys:
+                    log.error("Key %s does not exist on target %s", args.key, target)
+                    continue
```

Comment:
```suggestion
                    raise RegistryKeyNotFoundError()
```
We can piggy back on the error in the try/except.